File: boatpos-server/boatpos-server-service/boatpos-server-service-api/src/main/java/org/boatpos/api/dto/PromotionBefore.java
package org.boatpos.api.dto;

import com.google.gson.annotations.Expose;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;

/**
 * A {@link Promotion} that is handled before the rental.
 */
@SuppressWarnings("unused")
public class PromotionBefore extends Promotion {

    /**
     * The time-credit of this promotion in minutes.
     */
    @NotNull
    @Min(0)
    @Expose
    private Integer timeCredit;

    public PromotionBefore() {
    }

    public PromotionBefore(Long id, Integer version, String name, Integer timeCredit, String priceCalculation, Integer priority) {
        super(id, version, name, priceCalculation, priority);
        this.timeCredit = timeCredit;
    }

    public Integer getTimeCredit() {
        return timeCredit;
    }

    public void setTimeCredit(Integer timeCredit) {
        this.timeCredit = timeCredit;
    }
}

File: boatpos-client/src/promotionBefore.ts
export class PromotionBefore {
    id:number;
    name:string;
    timeCredit:number;
    enabled:boolean;
    priority:number;
    selected:boolean;
    keyBinding:string;

    constructor(id:number,
                name:string,
                timeCredit:number,
                enabled:boolean,
                priority:number,
                keyBinding:string) {
        this.id = id;
        this.name = name;
        this.timeCredit = timeCredit;
        this.enabled = enabled;
        this.priority = priority;
        this.keyBinding = keyBinding;
    }
}